### PR TITLE
[W5.11] [W11-A1] Wayne Neo

### PR DESCRIPTION
**Pull request by wzneo**
Second enhancement of 2KLoC work base
Addition of "Sort" command

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (ChengSashankh) [inline src/seedu/addressbook/parser/Parser.java:14]: It is better to import just the specific classes you need. If this happens automatically, you can change "code style -> Java -> imports" and change the setting

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+
+/**
+ * Lists all persons in the address book to the user.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/SortCommand.java:13]: Incorrect header message. The class is about `sort`, not `list`

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public CommandResult execute() {
+        List<Person> allPersons = addressBook.getUniquePersonList().getInternalList();
+        if (allPersons.size() == 0){
+            return new CommandResult(MESSAGE_EMPTY_LIST);
+        }
+        allPersons.sort(new AlphabeticalOrder());
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    private class AlphabeticalOrder implements Comparator<Person> {
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/SortCommand.java:36]: You can add header comments for this inner class.

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -128,4 +128,6 @@ public boolean equals(Object other) {
                         && this.allPersons.equals(((AddressBook) other).allPersons)
                         && this.allTags.equals(((AddressBook) other).allTags));
     }
+
+    public UniquePersonList getUniquePersonList(){ return allPersons;}
```
Comment (okkhoy) [inline src/seedu/addressbook/data/AddressBook.java:132]: 1. Please check the coding standards on how to write methods containing single line of code.
2. Need a space between `) {`

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -140,4 +140,6 @@ public boolean equals(Object other) {
                 || (other instanceof UniquePersonList // instanceof handles nulls
                         && this.internalList.equals(((UniquePersonList) other).internalList));
     }
+
+    public List<Person> getInternalList() { return internalList; }
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/UniquePersonList.java:144]: How to write methods containing a single line of code?

File: src/seedu/addressbook/parser/Parser.java (hunk 1 above)
Comment (okkhoy) [reply src/seedu/addressbook/parser/Parser.java:14]: +1

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 6]
```diff
@@ -12,17 +12,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (okkhoy) [inline test/java/seedu/addressbook/parser/ParserTest.java:15]: Don't import everything using *; import the necessary classes explicitly.

**Comment by okkhoy** [comment]
@AzuraAiR 
Some comments added. Please ack & close the PR after reading comments.

**Comment by wzneo** [comment]
Noted with thanks

**Pull request closed without merging**